### ISSUE TITLE
How to create bodySchema & other params

### ISSUE THREAD

**Issue by EwenQuim** [pb33f/libopenapi#261]
I'm trying to declare a bodySchema. Am I doing it right ?

I wonder if I'm using the `high` packages correctly. It should help and be "porcelain" as declared in the docs, but the NewXXX methods always require the use of `low` packages, which is disturbing.

```go
bodySchema, ok := s.OpenApiSpec.Model.Components.Schemas.Get(bodyTag)
if !ok {
	bodySchema := base.NewSchemaProxy(&low.NodeReference[*lowBase.SchemaProxy]{
		Value: &lowBase.SchemaProxy{
			Reference: low.Reference{},
		},
	})

	s.OpenApiSpec.Model.Components.Schemas.Set(bodyTag, bodySchema)
}
```

---

The docs specify that the lib can be used to load documents, but creating one from Go code seems very complicated as shown above.

I need to know if it's feasible before going further, are you supporting the possibility to iteratively create a spec from Go code ?

**Comment by daveshanley** [comment]
What you have written is correct. 

The model was initially built for reading, then we upgraded it to be writable, then we upgraded it again to be renderable and ordered. 

However, yes, in order to create a schema, you will need to use the plumbing APIs if you're making everything manually / programmatically. 

There could be some higher-level APIs added to remove the need to use the low-level APIs for sure, but yes, what you see here is a gap in the experience and shows you where we are with the model's maturity.
